src/m365/spo/commands/file/file-version-get.spec.ts

```diff
@@ -0,0 +1,210 @@
+    const actual = await command.validate({ options: { webUrl: validWebUrl, label: validLabel, fileId: validFileId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('retrieves version from a file with the fileUrl options', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${validWebUrl}/_api/web/GetFileByServerRelativeUrl('${formatting.encodeQueryParameter(validFileUrl)}')/versions/?$filter=VersionLabel eq '${validLabel}'`) {
+        return fileVersionResponse;
+      }
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: validWebUrl,
+        label: validLabel,
+        fileUrl: validFileUrl
+      }
+    });
+    assert(loggerLogSpy.calledWith(fileVersionResponse.value[0]));
+  });
+
+  it('retrieves version from a file with the fileId options', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${validWebUrl}/_api/web/GetFileById('${validFileId}')/versions/?$filter=VersionLabel eq '${validLabel}'`) {
+        return fileVersionResponse;
+      }
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: validWebUrl,
+        label: validLabel,
+        fileId: validFileId
+      }
+    });
+    assert(loggerLogSpy.calledWith(fileVersionResponse.value[0]));
+  });
+
+  it('properly escapes single quotes in fileUrl', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url = `${validWebUrl}/_api/web/GetFileByServerRelativeUrl('Shared%20Documents%2FFo''lde''r')/versions/?$filter=VersionLabel eq '${validLabel}'`) {
+        return fileVersionResponse;
+      }
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: false,
+        webUrl: validWebUrl,
+        label: validLabel,
+        fileUrl: `Shared Documents/Fo'lde'r`
+      }
+    });
+    assert(loggerLogSpy.calledWith(fileVersionResponse.value[0]));
+  });
+
+  it('command correctly handles version list reject request', async () => {
+    const err = 'Invalid version request';
+    sinon.stub(request, 'get').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/web/GetFileById') > -1) {
+        throw err;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: validWebUrl,
+        label: validLabel
+      }
+    }), new CommandError(err));
+  });
+
+  it('supports debug mode', () => {
+    const options = command.options;
+    let containsDebugOption = false;
+    options.forEach(o => {
+      if (o.option === '--debug') {
+        containsDebugOption = true;
+      }
+    });
+    assert(containsDebugOption);
+  });
+
+  it('supports specifying URL', () => {
```

Comment:
I generally think the more tests the better. But this one I think we may avoid 😉